test/system/220-healthcheck.bats

```diff
@@ -109,4 +73,40 @@ Log[-1].Output   | \"Uh-oh on stdout!\\\nUh-oh on stderr!\"
     run_podman rmi   healthcheck_i
 }
 
+@test "podman healthcheck --health-on-failure" {
+    ctr="healthcheck_c"
+    img="healthcheck_i"
+    _build_health_check_image $img cleanfile
+
+    for policy in restart;do
```

Comment:
Need to re-add the other actions. I removed them temporarily to debug the `restart` action.